plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/interactive/rectpack/SetAspectRatio.xtend

```diff
@@ -0,0 +1,26 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright 2020 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This code is provided under the terms of the Eclipse Public License (EPL).
+ */
+package de.cau.cs.kieler.klighd.lsp.interactive.rectpack
+
+import org.eclipse.xtend.lib.annotations.Data
+
+/**
+ * Data class to set a new aspect ratio which is sent from client to server for the rectpacking algorithm.
+ * 
+ * @author sdo
+ */
+@Data
+class SetAspectRatio {
+    String id
+    Double aspectRatio
```

Comment:
I guess this question is for yourself. I think this is due to the property ASPECT_RATIO in RectPackingOptions that cannot reference primitive types because of Java generics. That should however be castable on the fly so this won't need the Double type.
I currently can't test that as for whatever reason the whole interactive stuff is not working for me (or I am using it wrong, as I find no examples online)

Reply:
Look [here](https://github.com/eclipse/elk-models/tree/master/tests/layered/interactive_layout) for interactive options. Are you already using Elk 0.7.0? We should into this problem.